File: tags/NHI1-0.4/theLink/example/perl/Filter3.pl
#+
#§  \file       theLink/example/perl/Filter3.pl
#§  \brief      \$Id$
#§  
#§  (C) 2009 - NHI - #1 - Project - Group
#§  
#§  \version    \$Rev$
#§  \author     EMail: aotto1968 at users.berlios.de
#§  \attention  this software has GPL permissions to copy
#§              please contact AUTHORS for additional information
#§

use strict;
use Net::PerlMsgque;

$| = 1;

package Filter3;
use base qw(Net::PerlMsgque::MqS);

  sub Filter {
    my $ctx = shift;
    my $ftr = $ctx->ServiceGetFilter();
    $ftr->SendSTART();
    $ftr->SendBDY($ctx->ReadBDY());
    if ($ctx->ServiceIsTransaction()) {
      $ftr->SendEND_AND_WAIT($ctx->ServiceGetToken());
      $ctx->SendSTART();
      $ctx->SendBDY($ftr->ReadBDY());
    } else {
      $ftr->SendEND($ctx->ServiceGetToken());
    }
    $ctx->SendRETURN();
  }

  sub ServerSetup {
    my $ctx = shift;
    my $ftr = $ctx->ServiceGetFilter();
    $ctx->ServiceCreate("+ALL", \&Filter);
    $ftr->ServiceCreate("+ALL", \&Filter);
  }

  sub new {
    my $class = shift;
    my $ctx = $class->SUPER::new(@_);
    $ctx->ConfigSetName("filter");
    $ctx->ConfigSetServerSetup(\&ServerSetup);
    $ctx->ConfigSetFactory(sub {new Filter3()});
    return $ctx;
  }


package main;

  our $srv = new Filter3();
  eval {
    $srv->LinkCreate(@ARGV);
    $srv->ProcessEvent({wait => "FOREVER"});
  };
  if ($@) {
    $srv->ErrorSet($@);
  }
  $srv->Exit();

1;

File: tags/NHI1-0.4/theLink/example/java/Filter3.java
/**
 *  \file       theLink/example/java/Filter3.java
 *  \brief      \$Id$
 *  
 *  (C) 2009 - NHI - #1 - Project - Group
 *  
 *  \version    \$Rev$
 *  \author     EMail: aotto1968 at users.berlios.de
 *  \attention  this software has GPL permissions to copy
 *              please contact AUTHORS for additional information
 */
package example;
import javamsgque.*;

class Filter3 extends MqS implements IFactory, IServerSetup {

  public MqS Factory() {
    return new Filter3();
  }

  public void ServerSetup() throws MqSException {
    MqS ftr = ServiceGetFilter();
    ServiceCreate("+ALL", new Filter()); 
    ftr.ServiceCreate("+ALL", new Filter());
  }

  public class Filter implements IService {
    public void Service(MqS ctx) throws MqSException {
      MqS ftr = ServiceGetFilter();
      ftr.SendSTART();
      ftr.SendBDY(ReadBDY());
      if (ServiceIsTransaction()) {
	ftr.SendEND_AND_WAIT(ServiceGetToken());
	SendSTART();
	SendBDY(ftr.ReadBDY());
      } else {
	ftr.SendEND(ServiceGetToken());
      }
      SendRETURN();
    }
  }

  public static void main(String[] argv) {
    MqS.Init("java", "example.Filter3");
    Filter3 srv = new Filter3();
    try {
      srv.ConfigSetName("filter");
      srv.LinkCreate(argv);
      srv.ProcessEvent(MqS.WAIT.FOREVER);
    } catch (Throwable e) {
      srv.ErrorSet(e);
    } finally {
      srv.Exit();
    }
  }
}



